diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/IProcessParameterListener.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/IProcessParameterListener.java
index 958c621157..85b0553f5b 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/IProcessParameterListener.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/IProcessParameterListener.java
@@ -38,4 +38,11 @@ public interface IProcessParameterListener {
 	public default String validate(ProcessParameterPanel parameterPanel) {
 		return null;
 	}	
+
+	/**
+	 * Initialization of process parameter form
+	 * @param parameterPanel
+	 */
+	public void onInit(ProcessParameterPanel parameterPanel);
+
 }
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java
index 49efdd5036..f07e4f9258 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/apps/ProcessParameterPanel.java
@@ -405,6 +405,15 @@ public class ProcessParameterPanel extends Panel implements
 		if (hasFields) {
 			centerPanel.appendChild(rows);
 			dynamicDisplay();
+
+			if (m_processInfo.getAD_Process_ID() > 0) {
+				String className = MProcess.get(Env.getCtx(), m_processInfo.getAD_Process_ID()).getClassname();
+
+				List<IProcessParameterListener> listeners = Extensions.getProcessParameterListeners(className, null);
+				for(IProcessParameterListener listener : listeners)
+					listener.onInit(this);
+			}
+
 		} else
 			dispose();
 		return hasFields;
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/BankTransferParameterListener.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/BankTransferParameterListener.java
index e26c423d13..e01c69ba42 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/BankTransferParameterListener.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/BankTransferParameterListener.java
@@ -57,4 +57,6 @@ public class BankTransferParameterListener implements IProcessParameterListener
 		}
 	}
 
+	public void onInit(ProcessParameterPanel parameterPanel) {
+	}
 }
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/CreateTableParameterListener.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/CreateTableParameterListener.java
index c2a7e71a59..5ff4b0afda 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/CreateTableParameterListener.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/CreateTableParameterListener.java
@@ -46,5 +46,8 @@ public class CreateTableParameterListener implements IProcessParameterListener {
 				}				
 			}
 		}
+	}	
+
+	public void onInit(ProcessParameterPanel parameterPanel) {
 	}
 }
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/GenerateModelParameterListener.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/GenerateModelParameterListener.java
index d4cd01a300..6208b17987 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/GenerateModelParameterListener.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/GenerateModelParameterListener.java
@@ -65,5 +65,8 @@ public class GenerateModelParameterListener implements IProcessParameterListener
 				}
 			}
 		}
+	}	
+
+	public void onInit(ProcessParameterPanel parameterPanel) {
 	}
 }
diff --git a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/MFARegisterParameterListener.java b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/MFARegisterParameterListener.java
index 01a4f1bbdf..a812fc2cb7 100644
--- a/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/MFARegisterParameterListener.java
+++ b/org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/process/MFARegisterParameterListener.java
@@ -55,4 +55,6 @@ public class MFARegisterParameterListener implements IProcessParameterListener {
 		}
 	}
 
+	public void onInit(ProcessParameterPanel parameterPanel) {
+	}
 }
